utils/validations.tsx

```diff
@@ -518,31 +586,134 @@ export const getDualFinanceMerkleAirdropSchema = () => {
+        if (!form.treasury) {
+          return this.createError({
+            message: `Please select a treasury`,
+          })
+        }
+        const numAtomsInTreasury = new BN(
+          form.treasury.extensions.token.account.amount
+        ).toNumber()
+        if (numAtomsInTreasury < val) {
+          return this.createError({
+            message: `Not enough tokens`,
+          })
+        }
+        return true
+      }),
   })
 }
 
-export const getDualFinanceLiquidityStakingOptionSchema = () => {
+export const getDualFinanceLiquidityStakingOptionSchema = ({
+  form,
+}: {
+  form: any
+}) => {
   return yup.object().shape({
     optionExpirationUnixSeconds: yup
       .number()
-      .typeError('Expiration is required'),
-    numTokens: yup.number().typeError('Num tokens is required'),
+      .typeError('Expiration is required')
+      .test(
+        'expiration',
+        'Expiration must be a future unix seconds timestamp',
+        function (val: number) {
+          const nowUnixMs = Date.now() / 1_000;
+          // Primary goal is to catch users inputting ms instead of sec.
+          if (val > nowUnixMs * 10) {
+            return this.createError({
+              message: `Please select a valid unix seconds timestamp`,
+            })
+          }
+          if (val < nowUnixMs) {
+            return this.createError({
+              message: `Please select a time in the future`,
+            })
+          }
+          return true
+        }
+      ),
+    numTokens: yup.number().typeError('Num tokens is required')
+    .test('amount', 'amount', async function (val: number) {
+      if (!form.baseTreasury) {
+        return this.createError({
+          message: `Please select a treasury`,
+        })
+      }
+      const numAtomsInTreasury = new BN(
+        form.baseTreasury.extensions.token.account.amount
+      ).toNumber()
+      if (numAtomsInTreasury < val) {
+        return this.createError({
+          message: `Not enough tokens`,
+        })
+      }
+      return true
+    }),
     lotSize: yup.number().typeError('lotSize is required'),
     baseTreasury: yup.object().typeError('baseTreasury is required'),
     quoteTreasury: yup.object().typeError('quoteTreasury is required'),
     payer: yup.object().typeError('payer is required'),
   })
 }
 
-export const getDualFinanceStakingOptionSchema = () => {
+export const getDualFinanceStakingOptionSchema = ({
+  form,
+}: {
+  form: any
+}) => {
   return yup.object().shape({
-    soName: yup.string().required('Staking option name is required'),
-    userPk: yup.string().required('User pk is required'),
+    soName: yup.string().required('Staking option name is required')
+    .test('is-not-too-long', 'soName too long', (value) =>
+      value !== undefined && value.length < 32
+    ),
+    userPk: yup
+      .string()
+      .test('is-valid-address1', 'Please enter a valid PublicKey', (value) =>
+        value ? validatePubkey(value) : true
```

Comment:
can we also make sure the account has a balance of sol? while not required its a good sanity check for fat fingers